cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -219,6 +219,45 @@ if(builtin_xxhash)
   add_subdirectory(builtins/xxhash)
 endif()
 
+#---Check for ZSTD-------------------------------------------------------------------
+# Potentially, this if() block will enable builtin_zstd.
+if (NOT builtin_zstd)
+  message(STATUS "Looking for ZSTD")
+  find_package(ZSTD)
+  if (ZSTD_FOUND)
+    message(STATUS "System ZSTD found; using that.")
+  else()
+    message(STATUS "System ZSTD not found. Switching on builtin_zstd option")
+    set(builtin_zstd ON CACHE BOOL "" FORCE)
+  endif()
+endif()
+
+if (builtin_zstd)
+  set(zstd_version v1.4.0)
+  set(ZSTD_TARGET ZSTD)
+  message(STATUS "Building ZSTD version ${zstd_version} included in ROOT itself")
+  set(ZSTD_URL https://github.com/facebook/zstd/archive/v1.4.0.tar.gz)
+  set(ZSTD_SHA256 63be339137d2b683c6d19a9e34f4fb684790e864fee13c7dd40e197a64c705c1)
+  set(ZSTD_LIBRARIES ${CMAKE_BINARY_DIR}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}zstd${CMAKE_STATIC_LIBRARY_SUFFIX})
+  set(ZSTD_C_FLAGS "-fPIC -Wno-unused-variable -O3")
```

Comment:
Those flags are incompatible with the Microsoft compiler

Reply:
Fixed